[winging-it] Add Winging it Index

stacyk opened the pull request:
![](https://source.unsplash.com/featured/?cute,animal&afdafs)

## Description
Add an Archive page and single view show winging it shows

## Steps to test/reproduce
- View archive (should be at /wingingit as we advertised)
- View single show
- View the /tags archive, and see the Winging It featured tag

https://deploy-preview-378--oddleventy.netlify.app/wingingit
https://deploy-preview-378--oddleventy.netlify.app/tags
https://deploy-preview-378--oddleventy.netlify.app/2023/07/09/winging-it-01/

## Show me

Winging It List Page:

![landing](https://github.com/oddbird/oddleventy/assets/1581694/9ebe2bc5-e390-4fa4-8405-67916e12993e)


Winging It Episode Single:

![single](https://github.com/oddbird/oddleventy/assets/1581694/9db9a3d3-0096-4d79-bfde-535123f18465)

Special Tag Section: 

![hover-over-highlighted-tag](https://github.com/oddbird/oddleventy/assets/1581694/a910b66c-a99a-4b24-92bb-740c91aea8e7)


Shows up in Posts List: 

![Post-list](https://github.com/oddbird/oddleventy/assets/1581694/c8c6c13a-b39c-4f38-ad55-9bf4f76b708b)

stacyk commented on content/blog/wingingit/winging-it-02.md:
```diff
@@ -0,0 +1,34 @@
+  - htmx
+image:
+  src: winging-it/winging-it-2.jpg
+summary: |
+  htmx summary
+---
+
+## What we cover:
+
+- What is htmx
+- Common UI patterns implemented with htmx
+- The benefits of using HTMX
+- The kinds of projects that are best suited for htmx
+
+## Links:
+
+- [htmx Examples](https://htmx.org/examples/)
+- [htmx intro](https://htmx.org/docs/)
+- [Infinite scroll](https://htmx.org/examples/infinite-scroll/)
+- [Active search](https://htmx.org/examples/active-search/)
+- [When Should You Use Hypermedia?](https://htmx.org/essays/when-to-use-hypermedia/)
+- [A Real World React ➡️ htmx Port](https://htmx.org/essays/a-real-world-react-to-htmx-port/)
+- [Hypermedia systems](https://hypermedia.systems/introduction/)
```
No need reviewing this as it will be updated after the show is recorded.

stacyk commented on content/blog/wingingit/wingingit.11tydata.yaml:
```diff
@@ -0,0 +1,6 @@
+venue: Winging It
+url: https://www.youtube.com/@wingingitoddbird
```
Since venue is a location , it controls the meta prefix used on the byline ... so right now it say:

**by Miriam, Ed, & Stacy at Winging It on July 9, 2023**

But I think it would sound better to say 

**by Miriam, Ed, & Stacy on Winging It posted July 9, 2023**

I could change the venue name and it would say 

**by Miriam, Ed, & Stacy at The Winging It Channel on July 9, 2023**

Anyone have any strong feelings about this?

SondraE replied:
@stacyk I have a slight preference for 
by Miriam, Ed, & Stacy at The Winging It Channel on July 9, 2023

stacyk replied:
@jgerigmeyer you may want to check my work since I added the length to the post list card and the post detail. I left it a string incase we ever have longer episodes and wanted more flexibility in how we display that. We may want to remove the space between the time and the label?

![length-post-detail](https://github.com/oddbird/oddleventy/assets/1581694/9c812a1b-4ab0-4308-93f8-40798bdf68f1)
![length-post-list](https://github.com/oddbird/oddleventy/assets/1581694/00fe3c76-c1c3-485a-875c-1cbf40e6c1c9)

jgerigmeyer replied:
@stacyk Looks good! I pushed minor cleanup; up to you about the spacing?

stacyk commented on content/wingingit.md:
```diff
@@ -0,0 +1,17 @@
+title: Winging It
+date: 2023-07-09
+index: Winging It
+tags: []
```
Im not sure I need this now that I moved this md file out of blog.

stacyk reviewed (commented):
@SondraE @mirisuzanne @jgerigmeyer This is ready for comment/review when you have a member. I have added a number of comments inline with some explanation in this PR.

stacyk commented:
![](https://github.trello.services/images/mini-trello-icon.png) [WINGING IT: Create post template](https://trello.com/c/IbtpDWXo/211-winging-it-create-post-template)

jgerigmeyer commented on content/_includes/page/recents.njk:
```diff
@@ -3,14 +3,16 @@
   'Link': 'Links',
-  'Case Study': 'Case Studies'
+  'Case Study': 'Case Studies',
+  'Winging It': 'Episodes'
```
@stacyk I fixed this in https://github.com/oddbird/oddleventy/commit/6fb5bd9b3b32ed4eb329b031d189ba9eeb2ccd0c, simplified the logic a bit and made a similar change for the other types of posts (so that e.g. on `/tags/podcast/` it says "Podcasts" instead of "Posts").

SondraE commented on content/_includes/icons/winging-it-lines.svg:
@stacyk LGTM!

stacyk commented on content/blog/wingingit/winging-it-01.md:
```diff
@@ -0,0 +1,38 @@
+---
+title: CSS Cascade Layers - Winging It 1
```
![Screenshot 2023-07-13 at 3 32 01 PM](https://github.com/oddbird/oddleventy/assets/1581694/bc92e5d1-de52-46bc-bcf4-c14ece9ef1a3)

I didn't add length yet.

mirisuzanne commented on content/blog/wingingit/winging-it-01.md:
```diff
@@ -0,0 +1,39 @@
+---
+title: 1. CSS Cascade Layers
```
Right now the episode numbers seem a bit strange out of context. I could see a few options, like adding the word 'episode' (eg `CSS Cascade Layers (episode 1)`)? Or putting the episode number in the subtitle? Or making that it's own metadata, that only shows sometimes - like with the new `length` data?

stacyk replied:
@mirisuzanne We discussed the episode name in the post title... but I like the idea of having its own metadata.  https://github.com/oddbird/oddleventy/pull/378/files/e6d110695220449656fc83f90e5d2f7b15f472ab#r1260210525

stacyk replied:
@jgerigmeyer following up from our discussion during stand up, my proposal for the winging it specific byline is/was:

Episode Title
Winging It episode # _hosted by_ name, name and name _on_ July 20, 2023 | 60 mins

but now that @mirisuzanne mentioned in the review response that we could treat all podcasts, not our own in a similar way, then I think episode number wouldn't work as I am willing to bet that not all podcasts include a number. We could drop it in favor of something like:

Episode Title
_on_ Winging It _with (or featuring?)_ name, name and name _on (or posted?)_ July 20, 2023 | 60 mins

jgerigmeyer replied:
Added in https://github.com/oddbird/oddleventy/commit/09980562e9ee0dfae022d5464de9f84bd7e54405

mirisuzanne commented on content/blog/wingingit/winging-it-01.md:
```diff
@@ -0,0 +1,38 @@
+tags:
+  - CSS
+  - Cascade Layers
+action:
+  text: Watch on YouTube »
+  url: https://youtu.be/FwICaSE8iuY
```
I have some preference for the ease of an embed (which we do with other videos). Is there a way for us to add our own subscribe button, or encourage that even with an embed?

stacyk replied:
@mirisuzanne I believe we can add a subscribe button, but I will look into that more tomorrow.

mirisuzanne commented on content/blog/wingingit/winging-it-01.md:
```diff
@@ -0,0 +1,39 @@
+  text: Watch on YouTube »
+  url: https://youtu.be/FwICaSE8iuY
+image:
+  src: winging-it/winging-it-1.jpg
```
Might be nice to have a bit more uniqueness in these, specific to the episode? But it looks really great.

mirisuzanne replied:
It's very not-urgent. It's a great idea to have a template we can use without a lot of customization - might just be good (eventually) to have a way to slot in a screen-grab of a demo or something else from the video. Fine if we want to ship with something simpler, tho.

mirisuzanne commented on content/blog/wingingit/winging-it-01.md:
```diff
@@ -0,0 +1,38 @@
+- How we historically dealt with cascade/specificity problems (before layers)
+- What CSS Layers are and how they work
+- Why unlayered styles are prioritized
+- How to start using Layers within a project
+- Additional best practices when using Layers
```
no opinion.

mirisuzanne commented on content/blog/wingingit/winging-it-01.md:
```diff
@@ -0,0 +1,39 @@
+  we discussed what CSS Layers are and how you can use them in your project.
+---
+
+## What we cover:
```
Do we want time codes? Related: should we break the video into chapters, or nah?

stacyk replied:
eventually, yes. I would like to discuss how much time we'd want to allow for this as I am worried the amount of time I've spent getting this up and running is more than I think I expected at least. I think the first Episode is going to take way longer than things moving forward as we now have a lot of the base assets we need for each episode moving forward.

mirisuzanne commented on content/blog/wingingit/wingingit.11tydata.yaml:
```diff
@@ -0,0 +1,6 @@
+venue: The Winging It Channel
```
I think we could keep this short and simple:
```suggestion
venue: Winging It
```

mirisuzanne replied:
Ok. To me this seems like a broader concern about the language in the byline. This is not a new issue specific to our podcast, but could be applied to [most of the podcasts we've posted](https://www.oddbird.net/tags/podcast/). I'd prefer a solution that we like across all podcasts, without needing to change their names or add "The `<x>` Channel" or "The `<x>` Podcast" to each title.

jgerigmeyer replied:
@stacyk @mirisuzanne What do you think of the changes in https://github.com/oddbird/oddleventy/commit/09980562e9ee0dfae022d5464de9f84bd7e54405?

mirisuzanne replied:
works for me 👍🏼

stacyk replied:
Some turn out a bit strange... https://deploy-preview-378--oddleventy.netlify.app/2022/04/14/linkedin/ is his name  the name of the show?

stacyk replied:
Like this one might make more sense to say "on Design Systems Advocate" instead of "on Jina": 
https://deploy-preview-378--oddleventy.netlify.app/2020/05/05/jina-ama/

jgerigmeyer replied:
@stacyk I noticed that too, but I don't think it's any stranger than the existing version? https://www.oddbird.net/2022/04/14/linkedin/

stacyk replied:
I suppose that is true

mirisuzanne reviewed (approved):
🥳

jgerigmeyer commented on content/blog/wingingit/winging-it-01.md:
```diff
@@ -0,0 +1,45 @@
+
+{% import "embed.macros.njk" as embed %}
+
+{{ embed.figure(
+  data=media
+) }}
```
@stacyk Add a caption here?

stacyk replied:
I removed it intentionally since what I had was a duplicate of the post title and the screen reader will narrate the video title so I wasn't sure it caption is really needed here. 
![Screenshot 2023-07-20 at 2 29 57 PM](https://github.com/oddbird/oddleventy/assets/1581694/2fa3c9de-7610-47b4-a624-a36daac148f9)

jgerigmeyer reviewed (approved):
@stacyk Looks good to me! I think the missing items are:

- subscribe button
- unique per-episode images
- time codes and/or chapter breaks

I know you might be planning to add some of those later, so feel free to document in a new story and merge if you like?

SondraE reviewed (approved):
Great job!!

The only tiny thing I saw was in the Winging It list page header text:
"latest on UX Design" > "latest _in_ UX Design"

I think it would be fine to change that when we publish the next episode or with any other updates in Jonny's list.

stacyk commented:
> @stacyk Looks good to me! I think the missing items are:
> 
> * subscribe button
> * unique per-episode images
> * time codes and/or chapter breaks
> 
> I know you might be planning to add some of those later, so feel free to document in a new story and merge if you like?

I added chapters to the video now but didn't link the content or write those timestamps down in the text summary. I don't know if those should match but the timestamps are short titles and the text summary uses longer bullet points. 

I think the unique images will be something we can work on longer term and can update them any time. [I added a card](https://trello.com/c/TKvQ8boM/27-find-a-way-to-slightly-customize-per-episode-thumbnails-and-website-hero-images). I am more concerned with the images on the YouTube videos (thumbnail) looking more distinct than these posts on our site... but I agree it would be nice to have something a bit custom per episode. 

The subscribe code is below, but I don't know the best way to add it to the site. As far as placement goes, @SondraE where do you think it would be expected? I don't know if the top right where we sometimes show the Source / Docs links would work but also I can see it either underneath the show notes text or maybe under the video embed? 

```

<script src="https://apis.google.com/js/platform.js"></script>

<div class="g-ytsubscribe" data-channelid="UCUkHxN78y9On9YH1zd-aTGw" data-layout="full" data-count="hidden"></div>

``` 

Here's what it will look like. 
![Screenshot 2023-07-20 at 4 01 51 PM](https://github.com/oddbird/oddleventy/assets/1581694/98e3e5c0-9d2c-4cf2-aa1c-8bdb3869df8d)

We can hide the logo and the little "Winging it" title if we want
![Screenshot 2023-07-20 at 4 01 59 PM](https://github.com/oddbird/oddleventy/assets/1581694/caac7e62-bbd3-4cd4-b12d-e799d290ed65)

This is the code for that: 

```

<script src="https://apis.google.com/js/platform.js"></script>

<div class="g-ytsubscribe" data-channelid="UCUkHxN78y9On9YH1zd-aTGw" data-layout="default" data-count="hidden"></div>

```
